packages/tokens/src/shadows/addOverflowShadow.ts

```diff
@@ -0,0 +1,123 @@
+import { transparentize } from 'polished';
+
+import { css } from '@leafygreen-ui/emotion';
+import { Theme } from '@leafygreen-ui/lib';
+import { palette } from '@leafygreen-ui/palette';
+
+export const Side = {
+  Top: 'top',
+  Left: 'left',
+  Right: 'right',
+  Bottom: 'bottom',
+} as const;
+export type Side = (typeof Side)[keyof typeof Side];
+
+const BLUR_RADIUS = 16;
+const SHORT_SIDE_SIZE = 36;
+
+const shadowThemeColor: Record<Theme, string> = {
+  [Theme.Light]: palette.gray.dark1,
+  [Theme.Dark]: palette.black,
+};
+
+const shadowOffset: Record<Theme, number> = {
+  [Theme.Light]: 8,
+  [Theme.Dark]: 16,
+};
+
+const getPseudoElement = (side: Side, isInside: boolean) => {
+  if ((side === Side.Right || side === Side.Bottom) && !isInside)
+    return ':after';
+  if ((side === Side.Top || side === Side.Left) && isInside) return ':before';
+  return !isInside ? '::before' : '::after';
+};
+
+const getPositionStyles = (
+  side: Side,
+  isInside: boolean,
```

Comment:
nit: I know you're not using `inset`, but thoughts on `isInset` to match box-shadow? Though maybe this might not be the best name if we end up not using box-shadows down the line.